src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SPACE_SEPARATOR = " ";
+    private static final String COMMA_SEPARATOR = ",";
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int OPERATION_AMOUNT_INDEX = 1;
+    private static final String SUPPLY_NAME = "supply";
+    private static final String BUY_NAME = "buy";
+    private static final String RESULT_NAME = "result";
+    private static final String COMMA_SYMBOL = ",";
+    private static final String CAN_NOT_READ_FILE_INFO = "Can't read data from the file ";
+    private static final String CAN_NOT_WRITE_FILE_INFO = "Can't write data from the file ";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readFromFile(fromFileName);
+        String report = getReport(dataFromFile);
+        writeToFile(toFileName, report);
+    }
+
+    private String readFromFile(String fromFileName) {
+
```

Comment:
For the future, remove the first empty line in all classes/interfaces/methods where it is present
```suggestion
```

Reply:
Ok, I'll be. I haven't seen that